data/debug-cli/docker_debug.yaml

```diff
@@ -0,0 +1,166 @@
+command: docker debug
+short: Get a shell into any container or image.
+long: |-
+  > **Note**
+  >
+  > Docker Debug requires a [Pro, Team, or Business subcription](/subscription/details/).
+  > You must [sign in](/desktop/get-started/) to use this command.
+
+  Get an enhanced shell with additional tools into any container or image.
+  
+  The shell contains the following builtin commands:
+  - `install [tool1] [tool2]`: Add Nix packages from: https://search.nixos.org/packages.
+  - `uninstall [tool1] [tool2]`: Uninstall NixOS package(s).
+  - `entrypoint`: Print/lint/run the entrypoint.
+  - `builtins`: Show builtin commands.
+
+  For images and stopped containers, all changes are discarded when leaving the
+  shell. At no point, do changes affect the actual image or container. When
+  accessing running or paused containers, all filesystem changes are directly
+  visible to the container. The `/nix` directory is never visible to the actual
+  image or container.
+  
+usage: debug [OPTIONS] {CONTAINER|IMAGE}
+pname: docker
+plink: docker.yaml
+options:
+    - option: shell
+      value_type: shell
+      default_value: "auto"
+      description: "Select a shell. Supported: `bash`, `fish`, `zsh`, `auto`."
+      deprecated: false
+      hidden: false
+      experimental: false
+      experimentalcli: false
+      kubernetes: false
+      swarm: false
+    - option: command
+      shorthand: c
+      value_type: string
+      default_value: false
+      description: Evaluate the specified commands instead, passing additional positional arguments through $argv.
+      deprecated: false
+      hidden: false
+      experimental: false
+      experimentalcli: false
+      kubernetes: false
+      swarm: false
+    - option: host
+      value_type: string
+      default_value: false
+      description: "Daemon docker socket to connect to. E.g.: `ssh://root@example.org`, `unix:///some/path/docker.sock`"
+      deprecated: false
+      hidden: false
+      experimental: false
+      experimentalcli: false
+      kubernetes: false
+      swarm: false
```

Comment:
🤔 we already have a `--host` command on the base command. Why do we have another one here? Is it different, how do they co-exist?

Reply:
Ah! Bookmarked this PR Yesterday, and was about to ask the same; if we have options on both levels, this will be confusing;

```bash
docker --host unix:///var/run/docker.sock debug --host <something else> ...
```

(maybe this was intended / used when running the plugin standalone, so not as plugin?)

Reply:
yes, this duplication is intended (it's similar to the -n flag in kubectl)

From during the interviews I learned many people don't associate the top level --host flag with commands. They wouldn't know about it or not expect it there. This duplication intends to emphasise docker debug really is designed to work with --host flag. A lot of people asked for the remote case. That's also why it's important to extra have an example with the --host flag on this doc page.

Reply:
Hm.... not sure I'm happy with that; we'd now have `docker context`, `DOCKER_HOST`, `docker --host` _AND_ `docker debug --host`.

Note that this may have consequences with other logic in the CLI and elsewhere as well (where data should be sent, OTEL, etc), so we should be careful with this.

Reply:
I can also see this being confusing, e.g. use `docker ps` to list the container you want to debug, then `docker debug --host other-host` ? Wondering what the flow is in that case.

Reply:
Is the --host flag actually duplicated or did you just include it here for the purpose of illustrating an example? If it it's only to show the debug _works_ with --host, then I'd remove it from this table and include a section on how to use it with remote hosts, with the syntax `docker --host x debug y` (and with `context` as well if that also works). Then it would be clear that it's using the existing mechanisms. Again, if that's the intention here.

Reply:
I'm also wondering how this would work, as "remote" won't be a Docker Desktop install, but a regular engine, which doesn't have the services running for debug;

Here's my list of contexts; I have one for a remote engine;

```bash
docker context ls
NAME              DESCRIPTION                               DOCKER ENDPOINT                                   ERROR
default           Current DOCKER_HOST based configuration   unix:///var/run/docker.sock
desktop-linux *   Docker Desktop                            unix:///Users/thajeztah/.docker/run/docker.sock
remote                                                      ssh://swarm-test-01
```

Doing a quick check; it looks like the remote engine has a `hello-world` image;

```bash
docker --context=remote image ls
REPOSITORY                TAG       IMAGE ID       CREATED        SIZE
busybox                   latest    a416a98b71e2   6 months ago   4.26MB
alpine                    latest    9ed4aefc74f6   9 months ago   7.05MB
hello-world               latest    feb5d9fea6a5   2 years ago    13.3kB
```

Trying to debug the image on the remote machine, it pulls the `docker/desktop-docker-debug-service:0.0.18` image on the remote machine, but then ... also pulls the current version of the  `hello-world` image (probably the `hello-world` image on the remove machine was outdated), and then fails?

```bash
docker debug --host ssh://swarm-test-01 hello-world
...
30a05b81ac91: Pull complete
0b540ad1848e: Pull complete
Digest: sha256:d3dc0f309dc6bc8aa69c59dcc5316c84f14a065d8a06a2bfa2bb6d44be33242e
Status: Downloaded newer image for docker/desktop-docker-debug-service:0.0.18
latest: Pulling from library/hello-world
c1ec31eb5944: Pull complete
Digest: sha256:4bd78111b6914a99dbc560e6a20eab57ff6655aea4a80c50b0c5491968cbc2e6
Status: Downloaded newer image for hello-world:latest
Error: error during connect: Get "http://labs.debug-tools.localhost/lookup": unable to upgrade to tcp, received 404
```

```bash
REPOSITORY                            TAG       IMAGE ID       CREATED        SIZE
docker/desktop-docker-debug-service   0.0.18    5a76776fd058   3 days ago     1.59GB
busybox                               latest    a416a98b71e2   6 months ago   4.26MB
hello-world                           latest    d2c94e258dcb   8 months ago   13.3kB
alpine                                latest    9ed4aefc74f6   9 months ago   7.05MB
hello-world                           <none>    feb5d9fea6a5   2 years ago    13.3kB
```